src/h3lib/lib/edge.c

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2022 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @file  edge.c
+ * @brief Edge functions for manipulating (undirected) edge indexes.
+ */
+
+#include <inttypes.h>
+#include <stdbool.h>
+
+#include "algos.h"
+#include "constants.h"
+#include "coordijk.h"
+#include "h3Assert.h"
+#include "h3Index.h"
+#include "latLng.h"
+#include "vertex.h"
+
+/**
+ * Wrap the error code from a directed edge function and present
+ * undirected edge errors instead.
+ */
+H3Error wrapDirectedEdgeError(H3Error err) {
+    if (err == E_DIR_EDGE_INVALID) {
+        return E_UNDIR_EDGE_INVALID;
+    }
+    return err;
+}
+
+/**
+ * Allows for operations on an edge index as if it were a directed edge
+ * from the edge owner to the neighboring cell.
+ * @param edge Undirected edge index
+ * @return H3Index Directed edge index
+ */
+H3Index edgeAsDirectedEdge(H3Index edge) {
+    // Do not make indexes that are not edges look "valid".
+    if (H3_GET_MODE(edge) == H3_EDGE_MODE) {
+        H3_SET_MODE(edge, H3_DIRECTEDEDGE_MODE);
+    }
+    return edge;
+}
+
+/**
+ * Returns an edge H3 index based on the provided neighboring cells
+ * @param cell1 An H3 hexagon index
+ * @param cell2 A neighboring H3 hexagon index
+ * @param out Output: the edge H3Index
+ */
+H3Error H3_EXPORT(cellsToEdge)(H3Index cell1, H3Index cell2, H3Index *out) {
```

Comment:
Not a blocker on this PR, but there may be a `cellsToEdges` function that can take any arbitrary indexes and return an array of edges between them, similar to `gridPathCells`?

Reply:
I think that's intended to be named `gridPathEdges` (which is why `gridPathCells` has `Cells` in the name)